src/client/Microsoft.Identity.Client/Platforms/iOS/AuthenticationContinuationHelper.cs

```diff
@@ -57,8 +58,21 @@ public static bool IsBrokerResponse(string sourceApplication)
         /// <param name="url"></param>
         public static void SetBrokerContinuationEventArgs(NSUrl url)
         {
+            // url should always be present
+            // if coming from broker, will be the broker response
+            // check to prevent null ref later
+            if (url == null)
+            {
+                return;
+            }
+
             Debug.WriteLine("SetBrokercontinuationEventArgs Called with Url {0}", url);
-            iOSBroker.SetBrokerResponse(url);
+
+            string urlString = string.Format(CultureInfo.CurrentCulture, url.ToString());
+            if (urlString.Contains(iOSBrokerConstants.IdentifyiOSBrokerFromResponseUrl))
+            {
+                iOSBroker.SetBrokerResponse(url);
```

Comment:
I think this may lead to a deadlock. I don't understand under which circumstance this condition would not be hit but  but it looks to me that you start a broker flow, which acquires a semaphore, then when the continuation is hit, you may or may not release the semaphore. If you don't release the semaphore the app is blocked.

Reply:
discussed off-line

---
In reply to: [354580232](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1534#discussion_r354580232) [](ancestors = 354580232)